Description: Update check's configure.ac to handle modern build systems
Author: Jason Pereira <jason.pereira@sohonet.com>
Last-Update: 2024-03-18
Forwarded: no

--- a/deps/check-0.10.0/configure.ac
+++ b/deps/check-0.10.0/configure.ac
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.

 # Prelude.
-AC_PREREQ([2.59])
+AC_PREREQ([2.71])

 AC_INIT([Check], [0.10.0], [check-devel at lists dot sourceforge dot net])
 CHECK_MAJOR_VERSION=0
@@ -19,6 +19,9 @@
 # place to put some extra build scripts installed
 AC_CONFIG_AUX_DIR([build-aux])

+# Add hardening flags
+CFLAGS="$CFLAGS -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security"
+
 # really severe build strictness
 AM_INIT_AUTOMAKE([-Wall gnits 1.11.2])